docs/hooks.md

```diff
@@ -0,0 +1,44 @@
+# Gopass Hooks
+
+`gopass` exposes some hook-able events during it's invocation lifecycle. This allows uses to inject additional functionality of perform addition logging.
+
+## Hook API
+
+All hooks are subject to the follwing constraints:
+
+* Hooks do not inherit `STDIN` or `STDOUT` from the parent process.
+* Hooks do inherit `STDERR` from the parent process and may use it to print anything they want.
+* Hooks always run from the `password-store` directory.
+* Hooks are run with the `GOPASS_HOOK=1` in their environment and with `GOPASS_CONFIG_DIR` set to the configuration directory the original `gopass` command was started.
+* An exit from a hook (or execution failure) cases the entire `gopass` command to fail.
```

Comment:
```suggestion
* An exit from a hook (or execution failure) causes the entire `gopass` command to fail.
```

Reply:
What if it's a post-hook? Would the whole operation still fail and it reverts to the initial state?

Reply:
Good question. Some operations are hard to revert. I don't think I can easily manage that complexity without a transactional backend. So: no. The command will fail but everything that's been done will stay as it is.